core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -78,14 +78,17 @@ object RequestChannel extends Logging {
     val header: RequestHeader =
       if (requestObj == null) {
         buffer.rewind
-        RequestHeader.parse(buffer)
+        try RequestHeader.parse(buffer)
+        catch {
+          case ex: Throwable =>
+            throw new InvalidRequestException(s"Error parsing request header for apiKey: $requestId", ex)
```

Comment:
@gwenshap Yes, that's a good point. In the case I was seeing, I was getting an error because a field was missing (client_id). Including the request id made it clear that the problem was much worse (the request id was garbage as per your comment) and I then figured out that the wrong port was being used (SSL instead of SASL_SSL). I tweaked the message as per your suggestion.

If you think it helps, we could also check if the request id is actually valid by calling `ApiKeys.forId` and mention it if not.